R/postprocess.R

```diff
@@ -403,3 +378,74 @@ enw_quantiles_to_long <- function(posterior) {
   long[, quantile := as.numeric(quantile) / 100]
   return(long[])
 }
+
+#' Build the ord_obs `data.table`.
+#'
+#' @param obs Observations as pulled from nowcast$latest[[1]].
+#' @param max_delay Whole number representing the maximum delay
+#' in units of the timestep.
+#' @param timestep The timestep to be used. This can be a string
+#' ("day", "week", "month") or a numeric whole number representing
+#' the number of days.
+#' @param sample String, "get_sample" or "no_sample".
+#' @param nowcast If getting posterior samples, the fit to pull
+#' the draws from.
+#'
+#' @return ord_obs A `data.table`.
+#'
+#' @family postprocess
+
+build_ord_obs <- function(obs, max_delay, timestep, sample, nowcast = NULL) {
+    internal_timestep <- get_internal_timestep(timestep)
+    ord_obs <- coerce_dt(
+    obs, required_cols = c("reference_date", "confirm"), group = TRUE
+  )
+  check_timestep_by_group(
+    ord_obs, "reference_date", timestep, exact = TRUE
+  )
+
+  ord_obs <- subset_obs(ord_obs, max_delay, internal_timestep,
+                        select = "after")
+
+  data.table::setorderv(ord_obs, c(".group", "reference_date"))
+  if (sample == "get_sample") {
```

Comment:
or any character string inputs we should check them using `rlang::arg_match`. This would look like this.. 

```r
test <- function(test = c("option1", "option2")) {
	test <- rlang::arg_match(test)
}
```